### PR TITLE
Add PathOp for matching extra files and content

### LINKED ISSUE #83: fs: more PathOp for matching extra files and content
Right now there is `MatchAnyFileContent`, and `MatchExtraFiles` to match any content and any extra files, but we might want to only allow specific files, or specific content that isn't entirely predictable.

For those cases we could add:

```go

// MatchExtraFilesGlob is a PathOp that updates a Manifest to allow extra files in a directory which
// match the file glob pattern, and with properties that match all ops.
func MatchExtraFilesGlob(glob string, ops ...PathOp) PathOp {
}
```

`MatchFileContent` done in #133 
```go
// MatchFileContent is a PathOp that updates a Manifest to use the provided function to check a files
// content. The function should return true if the content is as expected.
func MatchFileContent(f func([]byte) bool) PathOp {
}
```

### PR DESCRIPTION
**Pull request by glefloch**
This PR add some `PathOp` for matching extra files and content. 

Closes #83 

Signed-off-by: glefloch <glfloch@gmail.com>

### CODE REVIEW

**Comment by glefloch** [comment]
@dnephin Thank you for the review. I'm going to rebase my branch and remove stuff related to glob matching. I will open another PR after.

File: fs/manifest.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ type file struct {
 	resource
 	content             io.ReadCloser
 	ignoreCariageReturn bool
+	assertFunc          func(b []byte) bool
```
[the remark this answers is not in the document]
Comment (glefloch) [reply fs/manifest.go]: Regarding the `CompareResult` interface, should I add it in the `assert/cmp` package or somewhere else in the `fs` package? Should I also create methods like `SuccessCompareResult`, `ComparaisonResultFailure` as it's done for the `Result` interface, to make it easier for users ?

File: fs/report_test.go [hunk 2]
```diff
@@ -176,3 +176,56 @@ func TestMatchAnyFileMode(t *testing.T) {
...
+	t.Run("content match", func(t *testing.T) {
+		manifest := Expected(t,
+			WithFile("data", "different", MatchFileContent(func(b []byte) bool {
+				return true
+			})))
+		assert.Assert(t, Equal(dir.Path(), manifest))
+	})
+	t.Run("content does not match", func(t *testing.T) {
+		manifest := Expected(t,
+			WithFile("data", "data", MatchFileContent(func(b []byte) bool {
```
[the remark this answers is not in the document]
Comment (glefloch) [reply fs/report_test.go]: Yes you right, I updated it.

**Comment by dnephin** [comment]
Thanks again! I pushed another to commit to this branch which exports `cmp.result` so that it can be used to implement `fs.CompareResult`.

**Pull request merged**